be/src/storage/lake/tablet_writer.h

```diff
@@ -72,6 +74,12 @@ class TabletWriter {
 
     // The total number of rows have been written.
     virtual int64_t num_rows() const = 0;
+
+    //  return the rowset txn meta generated by tablet writer
+    virtual RowsetTxnMetaPB* rowset_txn_meta() = 0;
+
+    // allow to set custom tablet schema for writer, used in partial update
+    virtual void set_tablet_schema(std::shared_ptr<const TabletSchema>& tschema) = 0;
```

Comment:
It's a little strange to define such a simple setter as virtual function. Maybe we can define `_schema` as a protected field of the base class.

Reply:
Done